Added Hangar Project

rlizzo opened the pull request:
<!-- 
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with 
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams.
Consider asking on our [Gitter channel](https://gitter.im/conda-forge/conda-forge.github.io)
if your recipe isn't reviewed promptly.
-->

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/hangar```) and found it was in an excellent condition.

rlizzo commented:
I confirm that I am willing to be listed as a recipe maintainer. 

Ping @lantiga to post confirmation as well.

chrisburr commented on recipes/hangar/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+  sha256: 85bbb1e8fe0cac56ca86257283a96472bd8410443d1b819d0e6b67f0a0d63214
+
+build:
+  number: 1
```
Start from 0, it only needs to changed if the previous build was deployed:
```suggestion
  number: 0
```

rlizzo replied:
fixed!

chrisburr commented on recipes/hangar/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+
+build:
+  number: 1
+  skip: true  # [py<36]
```
This looks like it could be noarch?
```suggestion
  noarch: python
```

chrisburr replied:
Oops I forgot to delete this comment, it can't be `noarch` due to the Windows selector.

rlizzo replied:
This project does not support anything less than `python 3.6`, (we use language features which aren't back ported to `2.7` or `3.4` / `3.5`).

Correct me if I'm wrong, but it's my understanding that `noarch` refers to a package which can be run on any system, and any python version? Since we don't support some python versions, It can't be `noarch`?

chrisburr replied:
`noarch` means the same "build" can be used with multiple platforms and python versions. You can introduce a constraint on the Python version without by putting `python >=3.6` and conda won't try to install it.

In general pure Python packages can be noarch but anything with compiled code can't as different binary formats are used on different platforms and the Python ABI changes between versions.

That said, a limitation of the current conda package format is that `noarch` packages have to always have the same dependencies so the `blosc-hdf5-plugin  # [not win]` line means this can't be made `noarch`.

chrisburr commented on recipes/hangar/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+requirements:
+  host:
+    - blosc-hdf5-plugin  # [not win]
+    - click
```
You probably don't need almost all of these in `host`? Most of the time just having `python` and `pip` is enough.

rlizzo replied:
fixed!

chrisburr commented on recipes/hangar/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+    - tqdm
+    - wrapt
+  run:
+    - blosc-hdf5-plugin  # [not win]
```
I guess this is just because `blosc-hdf5-plugin` isn't built for Windows? Would it be practical to make a pull request [here](https://github.com/conda-forge/blosc-hdf5-plugin-feedstock) so this can be `noarch: python`?

It's fine if you don't want to for whatever reason.

rlizzo replied:
So the `blosc-hdf5-plugin` library isn't a hard dependency for this project. It's just an optional package which mildly increases IO performance of compressed data.

It's actually a pretty huge achievement to make the `hdf5 blosc` filter available for Unix-like platforms alone. The `hdf5` plugin build/distribution/install system is fairly convoluted, and I'm not actively versed in its workings right now. It would take more time than I have right now to hack a window build script together. It might move up on my priority list in the future, but right now It's not feasible.

Since it's a totally optional package, i'm not worried about it being a major issue for distributing `Hangar` via `conda-forge`. 

Hopefully this is ok?

chrisburr replied:
Yeah that's fine, once @lantiga confirms I'll merge 👍

chrisburr commented on recipes/hangar/meta.yaml:
```diff
@@ -0,0 +1,77 @@
+
+about:
+  home: https://github.com/tensorwerk/hangar-py
+  license: Apache 2.0
```
Really minor but for consistency with the SPDX naming scheme:
```suggestion
  license: Apache-2.0
```

rlizzo replied:
Fixed!

chrisburr commented on recipes/hangar/meta.yaml:
```diff
@@ -0,0 +1,65 @@
+requirements:
+  host:
+    - pip
+    - python >=3.6
```
As this isn't `noarch` you can't use this way of containing the Python version. Instead you have one build per Python version per platform.
```suggestion
    - python
```

rlizzo replied:
fixed!

chrisburr commented on recipes/hangar/meta.yaml:
```diff
@@ -0,0 +1,65 @@
+    - msgpack-python >=0.6
+    - numpy
+    - protobuf
+    - python >=3.6
```
Same here:
```suggestion
    - python
```

rlizzo replied:
fixed!

rlizzo commented:
Thanks for the quick review @chrisburr, hopefully I've addressed all your concerns. let me know if there's anything else you need to know / want changed!

lantiga commented:
Hi @chrisburr, I confirm that I’m willing to be listed as a maintainer. Thanks for the support!
